packages/core/src/utils/i18n.ts

```diff
@@ -4,18 +4,21 @@ export { Resource };
 
 const DEFAULT_LANG = "nl";
 
-function getComponentClosestLanguage(el: HTMLElement): string {
-  let element: Element | null = el;
-  let closestElement: HTMLElement | null = element.closest("[lang]");
-
-  while (element && element.parentNode !== document && !closestElement) {
-    element = element.parentNode instanceof ShadowRoot ? element.parentNode.host : element.parentElement;
-    if (element !== null) {
-      closestElement = element.closest("[lang]");
-    }
+function getComponentClosestLanguage(element: HTMLElement): string {
```

Comment:
Waar ging dit op mis dan?

Reply:
Deze code was makkelijker te debuggen: Ik had bij elke route `console.log` staan, en bij de while was het moeilijker om te bepalen wat de aanleiding was.

En toen ik de diff uiteindelijk bekeek vond ik deze makkelijker te beredeneren, en ik vind het fraaier dat er maar één `.closest("[lang]")` query is.

Wat denk jij?